### ISSUE TITLE
Interface to render

### ISSUE THREAD

**Issue by kenn** [wmakley/simple_serializer#2]
I like the concept, but here's one thing that gives me a pause.

```ruby
render json: MyObjectSerializer.new(my_object).serializable_hash
```

I know where you're coming from,  but comparing that with the vanilla Rails way:

```ruby
render json: { user: user }
```

The current API looks overwhelming.

I usually abstract out `render json:` with `render_ok` and `render_error` to sprinkle metadata on it, and for that, it should take any objects.

```ruby
def render_ok(hash={})
  render json: { status: 'ok' }.merge(hash)
end

# render_ok user: @user   # => { status: 'ok', user: { ...} }
# render_ok users: @users # => { status: 'ok', users: [ ... ] }

def render_error(status, message = nil)
  render json: { status: status, message: message || status }, status: status
end

# render_error :bad_request, "invalid param: name" # => { status: 'bad_request', message: 'invalid param: name' }
```

If simple_serializer just overrides `as_json`, my code will work just fine. Otherwise, I can't see how I can keep this simplicity.

In other words, how can we add common structures to the JSON without adding a log of complexity?

**Comment by wmakley** [comment]
Yes, SimpleSerializer implements #as_json and #to_json. The example is more verbose than it needs to be.

I will not be implementing automatic serialization (inference of Serializer class) for `render json: @user`, because it was the source of half of my problems with active_model_serializers. KISS.

**Comment by wmakley** [comment]
Also, does that answer your question?

The verbosity becomes a major plus in a bigger codebase. It is critical to know for sure if it is using the correct serializer class, and verbosity speeds up understanding of the code rather than hiding the serializing fields in some magic somewhere.

**Comment by wmakley** [comment]
My issue with AMS was I could never be sure if it was using the Serializer classes. Sometimes it would, but if it didn't there wouldn't be an error, I'd just get totally unexpected JSON.

I'd really recommend fast_jsonapi, or jsonapi-rb over my gem. I just created this thing so I could ditch the horrific AMS memory leaks and over-engineered complexity, but I would use those or graphql if I were starting from scratch.
